From 8ef6683b11165b5e1fed0edaea071f9129341aa0 Mon Sep 17 00:00:00 2001
From: ctang <ctang@cloudera.com>
Date: Fri, 12 Aug 2016 15:23:22 -0400
Subject: [PATCH 0729/1164] CDH-41809 HIVE-14342: Beeline output is garbled
 when executed from a remote shell (Naveen Gangam,
 via Chaoyu Tang, reviewed by Mohit Sabharwal)

Change-Id: Icabebae8785675f06a21d63d72439befdc38555c
---
 bin/beeline    |    2 +-
 bin/ext/cli.sh |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/bin/beeline b/bin/beeline
index c23ab22..8f4484c 100644
--- a/bin/beeline
+++ b/bin/beeline
@@ -23,7 +23,7 @@ bin=`cd "$bin"; pwd`
 export HADOOP_USER_CLASSPATH_FIRST=true
 
 # If process is backgrounded, don't change terminal settings
-if [[ ! $(ps -o stat= -p $$) =~ "+" ]]; then
+if [[ ( ! $(ps -o stat= -p $$) =~ "+" ) && ! ( -p /dev/stdin ) ]]; then
   export HADOOP_CLIENT_OPTS="$HADOOP_CLIENT_OPTS -Djline.terminal=jline.UnsupportedTerminal"
 fi
 
diff --git a/bin/ext/cli.sh b/bin/ext/cli.sh
index ef3870c..554129b 100644
--- a/bin/ext/cli.sh
+++ b/bin/ext/cli.sh
@@ -24,7 +24,7 @@ fi
 
 updateBeelineOpts() {
   # If process is backgrounded, don't change terminal settings
-  if [[ ! $(ps -o stat= -p $$) == *+ ]]; then
+  if [[ ( ! $(ps -o stat= -p $$) =~ *+ ) && ! ( -p /dev/stdin ) ]]; then
     export HADOOP_CLIENT_OPTS="$HADOOP_CLIENT_OPTS -Djline.terminal=jline.UnsupportedTerminal"
   fi
 }
@@ -49,4 +49,4 @@ cli () {
 cli_help () {
   updateCli
   execHiveCmd $CLASS $JAR "--help"
-}
\ No newline at end of file
+}
-- 
1.7.9.5

